i3pystatus/wireguard.py

```diff
@@ -0,0 +1,81 @@
+from i3pystatus import IntervalModule
+from i3pystatus.core.command import run_through_shell
+
+__author__ = 'Pluggi'
+
+
+class Wireguard(IntervalModule):
+    """
+    Monitor Wireguard connections.
+
+    .. note::
+        You might want to add something like this to /etc/sudoers:
+
+        ``%wheel ALL = NOPASSWD: /bin/systemctl start wg-quick@wg0.service,/bin/systemctl stop wg-quick@wg0.service``
+
+    Formatters:
+
+    * {vpn_name} — Same as setting.
+    * {status} — Unicode up or down symbol.
+    * {output} — Output of status_command.
+    * {label} — Label for this connection, if defined.
+
+    """
+
+    color_up = "#00ff00"
+    color_down = "#FF0000"
+    status_up = '▲'
+    status_down = '▼'
+    format = "{vpn_name} {status}"
+
+    status_command = "systemctl is-active wg-quick@%(vpn_name)s"
+    vpn_up_command = "sudo /bin/systemctl start wg-quick@%(vpn_name)s.service"
+    vpn_down_command = "sudo /bin/systemctl stop wg-quick@%(vpn_name)s.service"
+
+    connected = False
+    label = ''
+    vpn_name = ''
+
+    settings = (
+        ("format", "Format string"),
+        ("color_up", "VPN is up"),
+        ("color_down", "VPN is down"),
+        ("status_down", "Symbol to display when down"),
+        ("status_up", "Symbol to display when up"),
+        ("vpn_name", "Name of VPN"),
+        ("vpn_up_command", "Command to bring up the VPN - default requires editing /etc/sudoers"),
+        ("vpn_down_command", "Command to bring up the VPN - default requires editing /etc/sudoers"),
+        ("status_command", "command to find out if the VPN is active"),
+    )
+
+    def init(self):
+        if not self.vpn_name:
+            raise Exception("vpn_name is required")
+
+    def toggle_connection(self):
+        if self.connected:
+            command = self.vpn_down_command
+        else:
+            command = self.vpn_up_command
+        run_through_shell(command % {'vpn_name': self.vpn_name})
+
+    def on_click(self, button, **kwargs):
+        self.toggle_connection()
+
+    def run(self):
+        command_result = run_through_shell(self.status_command % {'vpn_name': self.vpn_name})
```

Comment:
This seems needlessly complicated. Also, C-style format strings are not very common in Python in 2021. It would arguably be more user-friendly for the end-user to create a format string like so:

`systemctl is-active wg-quick@{vpn_name}`

Then to process the format string you can use `self.status_command.format(vpn_name=self.vpn_name)

This same pattern is repeated in a few places in this module with the different commands. I think they should all be changed to use the more user-friendly format strings.

Reply:
I just pushed a fix for it